### PR TITLE
Implement auto-completion support for Notifier scripts

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Review by tsteenbe — COMMENTED** [review]
@sschuberth  Am I right this auto-completion works similar to how you changed rules.kts, if you then shouldn't this PR also update https://github.com/oss-review-toolkit/ort/blob/master/examples/notifications.kts?

**Comment by sschuberth** [comment]
Yes, I'll also update the example once the details of this PR (like the chosen double-extension of notification scripts) are set. But I'll do this separately as this PR already has 8 commits.

**Comment by sschuberth** [comment]
> But I'll do this separately as this PR already has 8 commits.

Ok, I changed my mind an added 2 additional commits as I had to make a fixup anyway.

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ use):
 * [_Reporter_](#reporter) - presents results in various formats such as visual reports, Open Source notices or
   Bill-Of-Materials (BOMs) to easily identify dependencies, licenses, copyrights or policy rule violations.
 * [_Notifier_](./notifier) - sends result notifications via different channels (like
-  [emails](./examples/notifications.kts) and / or JIRA tickets).
+  [emails](./examples/notifications/src/main/resources/example.notifications.kts) and / or JIRA tickets).
```
Comment (mnonnenmacher) [inline README.md:57]: Commit message typo: "zu" -> "to"

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sschuberth) [reply README.md:57]: Fixed.

**Comment by mnonnenmacher** [comment]
> Yes, I'll also update the example once the details of this PR (like the chosen double-extension of notification scripts) are set.

I do agree with the chosen name.

**Review by mnonnenmacher — APPROVED** [review]

**Comment by sschuberth** [comment]
> I do agree with the chosen name.

I was considering to rather use `*.<toolname>.kts`, like `*.evaluator.kts` instead of `*.rules.kts`. That would al least "scale" if more tools got scripting. On the other hand it would get difficult again once a single tool supports multiple types of scripts...

**Pull request merged**